components/quickForm/quickForm.js

```diff
@@ -6,7 +6,45 @@ Template.quickForm.helpers({
   },
   innerContext: function quickFormContext() {
     var atts = this;
-    
+
+    // get schema
+    var schema = eval(this.collection).simpleSchema()._schema;
+
+    // if atts.fields exists, restrict the schema to specified fields
+    if (atts.fields) {
+      // note: haven't found a clean way to do this all in one go yet, so doing it later. 
+    }
+
+    // get list of unique field groups with any falsy values removed
+    // note: if atts.fields is specified, only consider fields contained in it
+    var fieldGroups = _.compact(_.unique(_.map(schema, function (property, key) {
+      return (!atts.fields || _.contains(atts.fields, key)) && property.autoform && property.autoform.group ;
```

Comment:
You will have to do `key = SimpleSchema._makeGeneric(key);` before this because otherwise `_.contains(atts.fields, key)` won't work when the key is something like `emails.0.address`, which is `emails.$.address` in the schema.